inference-engine/src/mkldnn_plugin/nodes/mkldnn_lrn_node.cpp

```diff
@@ -107,21 +102,57 @@ std::shared_ptr<MemoryDesc> MKLDNNLrnNode::getSrcMemDesc(mkldnn::primitive_desc_
     if (idx > 0) {
         return std::make_shared<CpuBlockedMemoryDesc>(getOriginalInputPrecisionAtPort(idx), getInputShapeAtPort(idx));
     } else {
-        return MKLDNNExtensionUtils::makeDescriptor(primitive_desc_it.dst_desc(idx));
+        if (getInputShapeAtPort(idx).isDynamic()) {
+            return MKLDNNExtensionUtils::makeUndefinedDesc(primitive_desc_it.src_desc(idx), getInputShapeAtPort(idx));
+        }
+        return MKLDNNExtensionUtils::makeDescriptor(primitive_desc_it.src_desc(idx));
     }
 }
 
 void MKLDNNLrnNode::createPrimitive() {
-    if (prim)
-        return;
+    if (inputShapesDefined()) {
+        if (needPrepareParams())
+            prepareParams();
+        updateLastInputDims();
+    }
+}
+
+void MKLDNNLrnNode::prepareParams() {
+    auto& srcMemPtr = getParentEdgeAt(0)->getMemoryPtr();
+    auto& dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
+    if (!srcMemPtr || !srcMemPtr->GetPrimitivePtr())
+        IE_THROW() << errorPrefix << " input memory did not allocate";
+    if (!dstMemPtr || !dstMemPtr->GetPrimitivePtr())
+        IE_THROW() << errorPrefix << "destination memory did not allocate";
+
+    const NodeDesc* selected_pd = getSelectedPrimitiveDescriptor();
+    if (selected_pd == nullptr)
+        IE_THROW() << errorPrefix << "preferable primitive descriptor did not set";
 
-    auto prim_desc = createPrimitiveDescriptor<mkldnn::lrn_forward::primitive_desc, mkldnn::lrn_forward::desc>();
+    auto inpDesc = getParentEdgeAt(0)->getMemory().GetDescWithType<DnnlMemoryDesc>();
+    const auto& in_candidate = inpDesc->getDnnlDesc();
+    mkldnn::algorithm alg = isAcrossMaps ? mkldnn::algorithm::lrn_across_channels : mkldnn::algorithm::lrn_within_channel;
```

Comment:
minor: seems to be const parameter and can be initialized no inside prepareParams()